File: CS/Demo.Module/BusinessObjects/IDocument.cs
using DevExpress.ExpressApp.DC;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.Validation;
using GenerateUserFriendlyId.Module.BusinessObjects;

namespace Demo.Module.BusinessObjects {
    [DefaultClassOptions]
    [DomainComponent]
    [XafDefaultProperty("Title")]
    [ImageName("BO_Note")]
    public interface IDocument : IUserFriendlyIdDomainComponent {
        [Calculated("Concat('D',PadLeft(ToStr(SequentialNumber),6,'0'))")]
        string DocumentId { get; }
        [RuleRequiredField("IDocument.Title.RuleRequiredField", DefaultContexts.Save)]
        [FieldSize(255)]
        string Title { get; set; }
        [FieldSize(8192)]
        string Text { get; set; }
    }
}

File: VB/Demo.Module/BusinessObjects/IDocument.vb
Imports DevExpress.ExpressApp.DC
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.Validation
Imports GenerateUserFriendlyId.Module.BusinessObjects

Namespace Demo.Module.BusinessObjects
	<DefaultClassOptions>
	<DomainComponent>
	<XafDefaultProperty("Title")>
	<ImageName("BO_Note")>
	Public Interface IDocument
		Inherits IUserFriendlyIdDomainComponent

		<Calculated("Concat('D',PadLeft(ToStr(SequentialNumber),6,'0'))")>
		ReadOnly Property DocumentId() As String
		<RuleRequiredField("IDocument.Title.RuleRequiredField", DefaultContexts.Save)>
		<FieldSize(255)>
		Property Title() As String
		<FieldSize(8192)>
		Property Text() As String
	End Interface
End Namespace
